README.md

```diff
@@ -0,0 +1,170 @@
+                           1, 2, 2, // l3cache grid
+                           1, 2, 2, // numa grid
+                           1, 2, 1, // socket grid
+                          16, 8, 8, // node grid
+    };
+```
+In terms of a standard MPI Cartesian communicator this would correspond to a `[64,64,32]` grid
+of ranks. Contrary to the MPI communicator, here it is possible to specify exactly how the
+in-node rank grid should look to achieve best speed and minimize off-node communication.
+
+## Resulting rank placement
+Looking at the above example, here is how the rank grid looks on a single compute node, on the socket level:
+```
+Level 3 HWCART_MD_SOCKET
+
+                      1   1   1   1
+                    1   1   1   1
+                  1   1   1   1
+                1   1   1   1
+              0   0   0   0
+            0   0   0   0
+          0   0   0   0
+        0   0   0   0
+
+                      1   1   1   1
+                    1   1   1   1
+                  1   1   1   1
+                1   1   1   1
+              0   0   0   0
+            0   0   0   0
+          0   0   0   0
+        0   0   0   0
+
+                      1   1   1   1
+                    1   1   1   1
+                  1   1   1   1
+                1   1   1   1
+              0   0   0   0
+            0   0   0   0
+          0   0   0   0
+        0   0   0   0
+
+                      1   1   1   1
+                    1   1   1   1
+                  1   1   1   1
+                1   1   1   1
+              0   0   0   0
+            0   0   0   0
+          0   0   0   0
+        0   0   0   0
+
+```
+In the above rank grid, all ranks with the same value are executed on the same memory domain - in this case socket.
+Here is how the distribution looks among NUMA nodes:
+```
+Level 2 HWCART_MD_NUMA
+
+                      7   7   7   7
+                    7   7   7   7
+                  6   6   6   6
+                6   6   6   6
+              3   3   3   3
+            3   3   3   3
+          2   2   2   2
+        2   2   2   2
+
+                      7   7   7   7
+                    7   7   7   7
+                  6   6   6   6
+                6   6   6   6
+              3   3   3   3
+            3   3   3   3
+          2   2   2   2
+        2   2   2   2
+
+                      5   5   5   5
+                    5   5   5   5
+                  4   4   4   4
+                4   4   4   4
+              1   1   1   1
+            1   1   1   1
+          0   0   0   0
+        0   0   0   0
+
+                      5   5   5   5
+                    5   5   5   5
+                  4   4   4   4
+                4   4   4   4
+              1   1   1   1
+            1   1   1   1
+          0   0   0   0
+        0   0   0   0
+```
+There are 8 NUMA nodes on this 2-socket system, and the printout above shows which ranks
+are bound to which NUMA node.
+
+## Cartesian communicator API
```

Comment:
This seems not to be the full API, e.g. create and finalize are missing. We should include those as well, together with input/output parameter description